src/Agent.Listener/Configuration/ServiceControlManager.cs

```diff
@@ -59,11 +67,13 @@ public void CalculateServiceName(AgentSettings settings, string serviceNamePatte
                     agentNameSubstring = StringUtil.SubstringPrefix(settings.AgentName, settings.AgentName.Length - exceededCharLength);
                 }
 
-                serviceName = StringUtil.Format(serviceNamePattern, accountNameSubstring, poolNameSubstring, agentNameSubstring);
-            }
+                string thirdPartServiceNameSubstring = poolNameSubstring ?? environmentNameSubstring;
```

Comment:
Nit: could it make sense to rename it to something like ResourceCollectionSubstring - to be more close to semantics? It's not always could be the third in service name (since there is a custom service name pattern)

Reply:
fixed